lib/ex_aws/sqs/client.ex

```diff
@@ -0,0 +1,122 @@
+defmodule ExAws.SQS.Client do
+  use Behaviour
+
+  @moduledoc """
+  The purpose of this module is to surface the ExAws.SQS API tied to a single
+  configuration chose, such that it does not need passed in with every request
+
+  Usage:
+  ```elixir
+  defmodule MyApp.SQS do
+    use ExAws.SQS.Client, otp_app: :my_otp_app
+  end
+  ```
+
+  In your config
+  ```elixir
+  config :my_otp_app, ExAws,
+    sqs: [], #SQS config goes here
+  ```
+  """
+
+  @type sqs_permission :: 
+    :all | 
+    :receive_message | 
+    :delete_message |
+    :change_message_visibility | 
+    :get_queue_attributes 
+  @type sqs_acl :: [{binary, sqs_permission}]
```

Comment:
That could definitely work, although you'd want simply `[{sqs_permission, binary},...]` without the  `| {sqs_permission, binary}`.

The former defines `[all: "account_id"]` and the latter `{:all, "account_id"}` which isn't a valid Keyword arg.

However, it looks like it's petty common to set multiple permissions for a given account id, and I think using the account_ids as the key is the right option. While keyword args are the standard for function options, Maps are perfectly appropriate for regular function args (See dynamo put item). With this in mind I think we would want something like:

```elixir
SQS. add_permission("queue", "label", %{"account1" => [:receive_message, : delete_message]})
```
This gives us a typespec of:
```elixir
@type sqs_acl :: %{ binary => sqs_permission | [sqs_permission, ...]}
```